testsuite/MDAnalysisTests/coordinates/test_netcdf.py

```diff
@@ -195,6 +201,393 @@ def test_dt(self, u):
         assert_almost_equal(ref, u.trajectory.dt, self.prec)
         assert_almost_equal(ref, u.trajectory.ts.dt, self.prec)
 
+    def test_box(self, u):
+        ref = np.array([0., 0., 0., 0., 0., 0.], dtype=np.float32)
+        for ts in u.trajectory:
+            assert_equal(ref, ts.dimensions)
+
+
+class TestNCDFReader3(object):
+    """NCDF trajectory with box, positions, forces and velocities
+
+    Added to address Issue #2323
+    """
+    prec = 3
+
+    coord_refs = np.array([
+        [[15.249873, 12.578178, 15.191731],
+         [14.925511, 13.58888, 14.944009],
+         [15.285703, 14.3409605, 15.645962]],
+        [[14.799454, 15.214347, 14.714555],
+         [15.001984, 15.870884, 13.868363],
+         [16.03358, 16.183628, 14.02995]]
+    ], dtype=np.float32)
+
+    frc_refs = np.array([
+        [[8.583388, 1.8023694, -15.0033455],
+         [-21.594835, 39.09166, 6.567963],
+         [4.363016, -12.135163, 4.4775457]],
+        [[-10.106646, -7.870829, -10.385734],
+         [7.23599, -12.366022, -9.106191],
+         [-4.637955, 11.597565, -6.463743]]
+    ], dtype=np.float32)
+
+    vel_refs = np.array([
+        [[-0.5301689, -0.16311595, -0.31390688],
+         [0.00188578, 0.02513031, -0.2687525],
+         [0.84072256, 0.09402391, -0.7457009]],
+        [[-1.7773226, 1.2307, 0.50276583],
+         [-0.13532305, 0.1355039, -0.05567304],
+         [-0.6182481, 1.6396415, 0.46686798]]
+    ], dtype=np.float32)
+
+    box_refs = np.array([
+        [28.81876287, 28.27875261, 27.72616397, 90., 90., 90.],
+        [27.06266081, 26.55555665, 26.03664058, 90., 90., 90.]
+    ], dtype=np.float32)
+
+    @pytest.fixture(scope='class')
+    def universe(self):
+        return mda.Universe(PRM_NCBOX, TRJ_NCBOX)
+
+    @pytest.mark.parametrize('index,expected', ((0, 0), (8, 1)))
+    def test_positions(self, universe, index, expected):
+        universe.trajectory[index]
+        assert_almost_equal(self.coord_refs[expected],
+                            universe.atoms.positions[:3], self.prec)
+
+    @pytest.mark.parametrize('index,expected', ((0, 0), (8, 1)))
+    def test_forces(self, universe, index, expected):
+        """Here we multiply the forces by 4.184 to convert from
+        kcal to kj
+        """
+        universe.trajectory[index]
+        assert_almost_equal(self.frc_refs[expected] * 4.184,
```

Comment:
Are the reference forces in Amber units kcal/mole/Å  and you do the manual conversion to see if MDA reads it correctly in MDA units?

Reply:
Yep, I essentially extracted the values manually from the NetCDF file as is. Since this is your typical _pmemd_ generated trajectory, energies are in kcal instead of kJ. I have added some extra comments in 54da919 to explain this more.